### PR TITLE
Add Error Handling To MuteIssueService

### LINKED ISSUE #5069: Add Error Handling To MuteIssueService

### PR DESCRIPTION
**Pull request by ugras-ergun-sonarsource**
Fixes #5069

### CODE REVIEW

File: src/Integration.Vsix/AsmRef_Integration.Vsix_Baseline_WithStrongNames.txt
Comment (ugras-ergun-sonarsource) [inline src/Integration.Vsix/AsmRef_Integration.Vsix_Baseline_WithStrongNames.txt:1]: This is the result of multiple commits no real change but due to nature of AsmRef we have files on the change.

File: src/ConnectedMode.UnitTests/ConnectedMode.UnitTests.csproj [hunk 1]
```diff
@@ -53,9 +53,9 @@
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.5.0" />
     <PackageReference Include="MSTest.TestAdapter" Version="1.4.0" />
     <PackageReference Include="MSTest.TestFramework" Version="1.4.0" />
-
     <PackageReference Include="Moq" Version="4.18.2" />
     <PackageReference Include="FluentAssertions" Version="5.9.0" />
+    <Reference Include="PresentationFramework" />
```
Comment (ugras-ergun-sonarsource) [inline src/ConnectedMode.UnitTests/ConnectedMode.UnitTests.csproj:58]: This package needed to be added for tests to build in 2019

Comment (georgii-borovinskikh-sonarsource) [reply src/ConnectedMode.UnitTests/ConnectedMode.UnitTests.csproj:58]: This is not a package tho. I don't understand it. Could you explain what is this in more detail?

File: src/ConnectedMode/Resources.resx [hunk 2]
```diff
@@ -289,4 +289,16 @@
...
+  <data name="MuteIssuesService_Error_Caption" xml:space="preserve">
+    <value>Error</value>
+  </data>
+  <data name="MuteIssuesService_Error_CommentAdditionFailed" xml:space="preserve">
+    <value>Issue is resolved but an error occured while adding the comment, please refer to the logs for more information.</value>
+  </data>
+  <data name="MuteIssuesService_Error_FailedToTransition" xml:space="preserve">
+    <value>Unable to resolve the issue, please refer to the logs for more information.</value>
+  </data>
+  <data name="MuteIssuesService_Error_InsufficientPermissions" xml:space="preserve">
```
Comment (georgii-borovinskikh-sonarsource) [inline src/ConnectedMode/Resources.resx:301]: I think there's specific name for the kind of permission level needed to be able to resolve issues on SQ/SC. We could add it to the error message so the user doesn't have to guess.

File: src/ConnectedMode/Transition/MuteIssuesService.cs [hunk 3]
```diff
@@ -41,25 +43,31 @@ internal class MuteIssuesService : IMuteIssuesService
...
         {
             this.activeSolutionBoundTracker = activeSolutionBoundTracker;
             this.logger = logger;
             this.muteIssuesWindowService = muteIssuesWindowService;
             this.threadHandling = threadHandling;
             this.sonarQubeService = sonarQubeService;
             this.serverIssuesStore = serverIssuesStore;
+            this.messageBox = messageBox;
+
+            rm = new ResourceManager(typeof(Resources));
```
Comment (georgii-borovinskikh-sonarsource) [inline src/ConnectedMode/Transition/MuteIssuesService.cs:70]: rm

File: src/Integration.Vsix/AsmRef_Integration.Vsix_Baseline_WithStrongNames.txt
Comment (georgii-borovinskikh-sonarsource) [reply src/Integration.Vsix/AsmRef_Integration.Vsix_Baseline_WithStrongNames.txt:1]: Could you unstage it then maybe?

**Review by georgii-borovinskikh-sonarsource — APPROVED** [review]

File: src/ConnectedMode.UnitTests/ConnectedMode.UnitTests.csproj (hunk 1 above)
Comment (ugras-ergun-sonarsource) [reply src/ConnectedMode.UnitTests/ConnectedMode.UnitTests.csproj:58]: I'll put it in another ITemGroup reduce to confusion

**Pull request merged**